src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREMENT_BY_HALF = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        this.values = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == values.length) {
+            resize();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == values.length) {
+            resize();
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else {
+            System.arraycopy(values, index, values, index + 1, size - index);
+            set(value, index);
```

Comment:
```suggestion
            values[index] = value;
```
in set() we check index once more - this is redundant

Reply:
done